packages/react/src/forms/RadioGroup/RadioGroup.test.tsx

```diff
@@ -0,0 +1,153 @@
+import React, {render, cleanup} from '@testing-library/react'
+import '@testing-library/jest-dom'
+import {axe, toHaveNoViolations} from 'jest-axe'
+
+import {RadioGroup} from './RadioGroup'
+import {Radio} from '../Radio'
+import {FormControl} from '../FormControl'
+
+expect.extend(toHaveNoViolations)
+
+describe('RadioGroup', () => {
+  afterEach(cleanup)
+
+  it('renders a radio group correctly into the document', () => {
+    const {getByLabelText, getByRole, getByText} = render(
+      <RadioGroup>
+        <RadioGroup.Label>Choices</RadioGroup.Label>
+        <RadioGroup.Caption>You can only pick one</RadioGroup.Caption>
+        <FormControl>
+          <FormControl.Label>Choice one</FormControl.Label>
+          <Radio name="demo" value="one" defaultChecked />
+        </FormControl>
+        <FormControl>
+          <FormControl.Label>Choice two</FormControl.Label>
+          <Radio name="demo" value="two" />
+        </FormControl>
+        <FormControl>
+          <FormControl.Label>Choice three</FormControl.Label>
+          <Radio name="demo" value="three" />
+        </FormControl>
+
+        <RadioGroup.Validation variant="success">Great job!</RadioGroup.Validation>
```

Comment:
There's a Jest warning related to validation prop in the test logs. Possible to take a look at fixing that please?

https://github.com/primer/brand/actions/runs/12053705065/job/33609998572?pr=830#step:10:53

![Screenshot 2024-11-27 at 16 59 11](https://github.com/user-attachments/assets/b80b3b64-2d6c-4169-92f0-f8682b937073)


(Doesn't seem to be happening for CheckboxGroup FWIW)

Reply:
Ah yea, thanks for raising this. I spotted this the other day and I meant to dig into it more.

It boils down to the fact that, in `FormControl.tsx`, we forward `validationStatus` to the Radio component, but Radio doesn't implement it, so it just ends up in the HTML.

I'm not sure if I can just stop FormControl from forwarding `validationStatus` to Radio, or whether Radio should actually be using it to set `aria-invalid` as Checkbox does.

My inclination is to tackle this in its own PR as it's not an issue introduced in this PR, wdyt?

Reply:
Thinking about it though, I'm not entirely sure how an individual Radio control (not a group, just a single control) could ever be _invalid_ 🤔

Reply:
I've removed the prop forwarding and resolved the `console.error` in the tests. Removing the forwarding hasn't modified the behaviour of FormControl or Radio in any way as the forwarded prop was never used.

8e910131f7366ebb0adc1ea3c1d3e9f262c52ef6